core/src/lib.rs

```diff
@@ -110,13 +108,57 @@ mod tests {
     }
     impl DocumentBuilder for TestObj {}
 
-    #[test]
-    fn test_create_credential() -> Result<(), String> {
-        let to = TestObj::new();
+    fn get_body_subject() -> (HashMap<String, Value>, HashMap<String, Value>) {
         let mut kv_body: HashMap<String, Value> = HashMap::new();
         let mut kv_subject: HashMap<String, Value> = HashMap::new();
 
-        let expect = json!({
+        kv_body = HashMap::from([
+            ("issuer", "did:example:28394728934792387"),
+            ("identifier", "83627465"),
+            ("name", "Permanent Resident Card"),
+            (
+                "description",
+                "Government of Example Permanent Resident Card.",
+            ),
+            ("issuanceDate", "2019-12-03T12:19:52Z"),
+            ("expirationDate", "2029-12-03T12:19:52Z"),
+        ])
+        .into_iter()
+        .map(|(k, v)| (k.into(), v.into()))
+        .collect();
+
+        kv_body
+            .entry("type".to_string())
+            .or_insert(json!(["VerifiableCredential", "PermanentResidentCard"]));
```

Comment:
you can use [`HashMap::insert`](https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#method.insert) for this one. Should be cleaner than `.entry(...).or_insert(...)`